api/xds_protocol.rst

```diff
@@ -155,45 +266,112 @@ the resources in the :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>`
+
+For :ref:`Listener <envoy_api_msg_Listener>` and :ref:`Cluster <envoy_api_msg_Cluster>` resource types, because each response
+needs to include all resources requested by the client, if a client requests a resource that does not exist, it can immediately
+tell this from the response.
+
+However, for other resource types, because each resource can be sent in its own response, there is no way to know from the
+next response whether the newly requested resource exists, because the next response could be an unrelated update for another
+resource that had already been subscribed to previously. As a result, clients are expected to use a timeout (recommended
+duration is 15 seconds) after sending a request for a new resource, after which they will consider the requested resource to
+not exist if they have not received the resource. In Envoy, this is done for :ref:`RouteConfiguration
+<envoy_api_msg_RouteConfiguration>` and :ref:`ClusterLoadAssignment <envoy_api_msg_ClusterLoadAssignment>` resources during
+:ref:`resource warming <xds_protocol_resource_warming>`.
+
+Note that even if a requested resource does not exist at the moment when the client requests it, that resource could be
+created at any time. Management servers must remember the set of resources being requested by the client, and if one of
+those resources springs into existence later, the server must send an update to the client informing it of the new resource.
+Clients that initially see a resource that does not exist must be prepared for the resource to be created at any time.
+
+Unsubscribing From Resources
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the incremental protocol variants, resources can be unsubscribed to via the :ref:`resource_names_unsubscribe
+<envoy_api_field_DeltaDiscoveryRequest.resource_names_unsubscribe>` field.
+
+In the SotW protocol variants, unsubscriptions are more complicated. For any request where the :ref:`resource_names
+<envoy_api_field_DiscoveryRequest.resource_names>` field is specified, a resource may be unsubscribed to by sending a new
+request without the resource name that the client wishes to unsubscribe from. For example, if the client had previously been
+subscribed to resources A and B but then sends a new request where the resource_names field contains only resource A, that will
+unsubscribe from B.
+
+For :ref:`Listener <envoy_api_msg_Listener>` and :ref:`Cluster <envoy_api_msg_Cluster>` resource types, if the initial
+request did not specify any resources to subscribe to, the management server uses site-specific wildcard behavior, as
+described above. However, if the initial request contains at least one resource name, then the management server knows that
+the client is not using the wildcard behavior. If a future request on the same stream sets the list of resource names to be
+empty, then the server must interpret this as unsubscribing from all resources of the specified type, rather than reverting
```

Comment:
@mattklein123 @lizan are you folks alright with this? This is the main new behavior being introduced in this PR, I would like to have this bounced off more than one person. FTR I'm good with it.

Reply:
This points to outdated text. Can you clarify what exactly we should review?

Reply:
I believe that Harvey was referring specifically to the idea of "wildcard" mode, which provides a way for gRPC clients (which do not use wildcard LDS/CDS queries) to unsubscribe from all LDS/CDS resources rather than falling back to wildcard behavior.  Please see the "How the client specifies what resources to return" and "Unsubscribing From Resources" sections for the relevant text.